### PR TITLE
Make FileSystemRazorProject internal

### PR DESCRIPTION
**Pull request by Yves57**
See #1117.

### CODE REVIEW

**Comment by pranavkm** [comment]
Probably need to make https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNetCore.Razor.Evolution/FileSystemRazorProjectItem.cs#L11 internal too

**Comment by Yves57** [comment]
@pranavkm I agree, but the specific property `FileSystemRazorProjectItem.File` is used in [RazorPageGenerator](https://github.com/aspnet/Razor/blob/dev/src/RazorPageGenerator/Program.cs#L133). Maybe replace this line by `var basePath = Path.GetDirectoryName(_source.PhysicalPath);`?

**Comment by pranavkm** [comment]
Sure, that should suffice.

File: test/Microsoft.AspNetCore.Razor.Evolution.Test/FileSystemRazorProjectTest.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public void NormalizeAndEnsureValidPath_NormalizesToForwardSlashes()
         public void EnumerateItems_DiscoversAllCshtmlFiles()
         {
             // Arrange
-            var fileSystemProject = new FileSystemRazorProject(TestFolder);
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Razor.Evolution.Test/FileSystemRazorProjectTest.cs]: It's better to call the constructor here. We want to make sure that we're really testing this class.

File: src/RazorPageGenerator/Program.cs [hunk 2]
```diff
@@ -130,7 +130,7 @@ public override Stream Read()
 
             private string ProcessFileIncludes()
             {
-                var basePath = _source.File.DirectoryName;
+                var basePath = System.IO.Path.GetDirectoryName(_source.PhysicalPath);
```
Comment (pranavkm) [inline src/RazorPageGenerator/Program.cs]: Does it conflict with the property name?

Comment (Yves57) [inline src/RazorPageGenerator/Program.cs]: Yes

**Comment by Yves57** [comment]
Updated PR.
Strange thing: I had to change the RazorPageGenerator target framework from `netcoreapp1.0` to `netcoreapp2.0` to be able to run it on my computer.

**Comment by pranavkm** [comment]
> Strange thing: I had to change the RazorPageGenerator target framework from netcoreapp1.0 to netcoreapp2.0 to be able to run it on my computer.

You might not have the 1.1.0 shared runtime installed. Unfortunately the package is used as a tool and there are issues that need to be worked out before it can start targeting netcoreapp2.0.

**Comment by Yves57** [comment]
@pranavkm The 1.1.0 runtime seems to be installed on my computer. Anyway, if it is only on my computer, it is not important 😃. Don't waste your time.
It was just to be sure that there is no bug in the RazorPageGenerator project.

File: src/Microsoft.AspNetCore.Razor.Evolution/RazorProject.cs [hunk 3]
```diff
@@ -123,5 +123,15 @@ protected virtual string NormalizeAndEnsureValidPath(string path)
 
             return path;
         }
+
+        /// <summary>
+        /// Create a Razor project based on a physical file system.
+        /// </summary>
+        /// <param name="rootDirectoryPath">The directory to root the file system at.</param>
+        /// <returns></returns>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Razor.Evolution/RazorProject.cs]: @Yves57 if you're around, could you fix this to say `The <see cref="RazorProject" />`. I can touch this up prior to merging it if I don't hear from you.

Comment (Yves57) [inline src/Microsoft.AspNetCore.Razor.Evolution/RazorProject.cs]: Done

**Comment by pranavkm** [comment]
Thanks for the PR!

**Pull request merged**